flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -646,7 +646,9 @@ Fortran::lower::createStringLiteral(Fortran::lower::FirOpBuilder &builder,
 fir::ExtendedValue Fortran::lower::componentToExtendedValue(
     Fortran::lower::FirOpBuilder &builder, mlir::Location loc,
     const fir::ExtendedValue &obj, mlir::Value component) {
-  auto fieldTy = fir::dyn_cast_ptrEleTy(component.getType());
+  auto fieldTy = component.getType();
+  if (auto ty = fir::dyn_cast_ptrEleTy(fieldTy))
+    fieldTy = ty;
```

Comment:
Is this actually being called both when `component` has a reference to a field and when `component` is a field value? The prevailing assumption was that fields would be in memory, so it's not clear when they are (register) values or if/when they are there isn't some deeper bug to address.